tools/automate_gcp_os_version.py

```diff
@@ -0,0 +1,118 @@
+"""
+checks for the latest os and updates the os versions in vars.tf file in dc-only, 
+multi-region and single connector deployments.
+Use this command to run this script in cloud shell: python3 tools/update_gcp_os.py
+"""
+import argparse
+import datetime
+import os
+import re
+import subprocess
+from subprocess import CalledProcessError
+
+COMPUTE_SERVICES_ENABLE = "gcloud services enable compute.googleapis.com"
+COMMANDS = ["gcloud compute images list"]
+PATHS = [["single-connector", "vars.tf"],
+         ["multi-region", "vars.tf"],
+         ["dc-only", "vars.tf"]]
+
+# commands = ["gcloud compute images list | grep windows-server-2016",
+#             'gcloud compute images list | grep centos-7',
+#             'gcloud compute images list | grep ubuntu-1804-bionic']
+
+
+# Added env variable for future purpose
+def get_os(env='cloud'):
+    """
+    get the latest version from GCP
+    """
+    try:
+        recent_os = []
+        if env == "cloud":
+            # breakpoint()
+            for command in COMMANDS:
+                capture_output = subprocess.run(command.split(' '), check=True,
+                                                capture_output=True).stdout.decode().split('\n')
+                recent_os.extend([key[0] for key in map(lambda x: x.split(' '), capture_output)])
+        return recent_os
+    except (AttributeError, TypeError, CalledProcessError) as ex:
+        print("Exception occurred: ", ex)
+
+
+def compare_versions(recent_os, line):
+    """
+    Checks for version numbers and returns true if latest version available
+    """
+    try:
+        # version_date variable is the current os version in the vars.tf
+        version = re.compile(r'[0-9]{8}').search(line.split("/")[-1])
+        if version:
+            version = version.group()
+            # print("version", version)
+            version_date = datetime.datetime.strptime(version, "%Y%m%d")
+            for _os in recent_os:
```

Comment:
now instead of going over the whole list of recent_os, you can use filter() and startswith() with version.group(1) to narrow down to the exact os you want.

You can then re-use the re.compile() object above to find the version number.  In fact, that could be put into a sub-function so that you can call something like get_version_date('windows-server-2016-dc-v20191210') get have it return the datetime object you want for comparison.